### PR TITLE
Visual Overhaul using BEM naming convention

### PR DESCRIPTION
**Pull request by Twiistrz**
Visual Overhaul using BEM naming convention.

<strong>Why BEM (Block, Element, Modifier)?</strong> The name of a BEM entity is unique. The same BEM entity always has the same name in all technologies (CSS, JavaScript, and HTML). The primary purpose of the naming convention is to give names meaning so that they are as informative as possible for the developer. [Learn more](https://en.bem.info/methodology/naming-convention/)

- ~~Redesign some components.~~
- ~~index.php new classes~~
    + ~~"main-filesource"~~
    + ~~"main-footer"~~
    + ~~"main-footer__warning"~~
    + ~~"main-footer__extension"~~
- ~~info_v3.js reword "Synopses" to "Synopsis"~~
- ~~index_v3.js change width 640 to 642~~

### CODE REVIEW

**Comment by soruly** [comment]
Why change width from 640px to 642px?

**Comment by Twiistrz** [comment]
I thought there will be some conflict on display "aesthetic". Well I don't see any problem, it can stay at 640 I guess.

image

**Comment by Twiistrz** [comment]
Visual Overhaul

Home:
image

About:
image
image

Changelog:
image

FAQs:
image
image
image

Terms:
image

File: css/index.css [hunk 1]
```diff
@@ -75,6 +83,7 @@ td:nth-child(3) {
 #preview {
     display: block;
     cursor: default;
+    width: 642px;
```
Comment (soruly) [inline css/index.css]: @Twiistrz did you forget to switch it back to 640px?

File: css/index.css [hunk 2]
```diff
@@ -182,20 +182,20 @@ td:nth-child(3) {
...
+
 #main {
     position: relative;
     width: 642px;
     margin-bottom: 1em;
     background-color: rgba(255, 255, 255, 1);
     color: #333;
-    box-shadow: 1px 1px 10px rgba(0, 0, 0, 0.3);
-    font-family: Helvetica, Arial, Meiryo, "Microsoft JhengHei", "Microsoft YaHei", sans-serif;
+    box-shadow: rgba(0, 0, 0, 0.25) 0px 1px 3px 0px;
```
[the remark this answers is not in the document]
Comment (soruly) [reply css/index.css]: @Twiistrz

[the remark this answers is not in the document]
Comment (Twiistrz) [reply css/index.css]: Its okay, but i can swap it out. Working on it.

File: css/index.css (hunk 1 above)
Comment (Twiistrz) [reply css/index.css]: That one is okay. When i use 640 there's a white space hanging around at the right side.

642px:
image

640px:
image

Whitespace:
image

Comment (Twiistrz) [reply css/index.css]: 642 should be fit perfectly, i guess

**Comment by Twiistrz** [comment]
Some screenshot on new color.  Will be working next on dark mode switcher.

image
image
image

**Comment by Twiistrz** [comment]
New mobile navigation v1

Mobile Navigation

**Comment by Twiistrz** [comment]
Update: Now using BEM naming convention

The new design: 
https://i.gyazo.com/358ee3192a267f39de3a72a20356bb3c.gif
trace moe overview design

Per page:
1_home
2_about
3_changelog
4_faq
5_terms

**Comment by soruly** [comment]
Have you tried the search results layout? with anilist panel and thumbnail images previews

**Comment by soruly** [comment]
> Update: Now using BEM naming convention
> 
> The new design:
> https://i.gyazo.com/358ee3192a267f39de3a72a20356bb3c.gif
> trace moe overview design
> 
> Per page:
> 1_home
> 2_about
> 3_changelog
> 4_faq
> 5_terms

Preferably I'd like to use the same color theme for all pages, like youtube is red in all pages, facebook is blue for all its pages.

**Comment by soruly** [comment]
It seems your css\app.css uses CSS variables which Internet Explorer 11 does not support
Can you test your layout on IE?

**Comment by Twiistrz** [comment]
Sure i'll try it out

**Comment by Twiistrz** [comment]
> 
> 
> It seems your css\app.css uses CSS variables which Internet Explorer 11 does not support
> Can you test your layout on IE?

Working on fallback version now.

**Comment by Twiistrz** [comment]
> 
> 
> Have you tried the search results layout? with anilist panel and thumbnail images previews

I can but it will take some time since all elements are in info_v3.js 

PS: No thumbnail I just copy the source from trace.moe (Its the same after all)
image
image
image

**Comment by soruly** [comment]
> > Have you tried the search results layout? with anilist panel and thumbnail images previews
> 
> I can but it will take some time since all elements are in info_v3.js
> 
> PS: No thumbnail I just copy the source from trace.moe (Its the same after all)
> image
> image
> image

It seems to me the top nav bar is too thick, taking too many vertical space. And the recent updates notification bar... seems occupying too much space too. But I'm not sure if there're better place to put it. Do you have better ideas?

**Comment by Twiistrz** [comment]
> 
> 
> > > Have you tried the search results layout? with anilist panel and thumbnail images previews
> > 
> > 
> > I can but it will take some time since all elements are in info_v3.js
> > PS: No thumbnail I just copy the source from trace.moe (Its the same after all)
> > image
> > image
> > image
> 
> It seems to me the top nav bar is too thick, taking too many vertical space. And the recent updates notification bar... seems occupying too much space too. But I'm not sure if there're better place to put it. Do you have better ideas?

working on it

**Comment by Twiistrz** [comment]
Well I just adjust the thickness after all.
image

**Comment by Twiistrz** [comment]
Style update for search, results list, info.

Search: Search
image

**Comment by Twiistrz** [comment]
Update stylesheet to LESS (Includes non minified and minified version of CSS).
Update classes, design, texts, indentions, divs
Add terms in sitemap.txt
Change css links to app.min.css

image

**Comment by Twiistrz** [comment]
Ready to merge anytime. If there's any problem or bug you can contact me anytime. Thank you.

Closing pull request.

**Pull request closed without merging**